packages/repository/src/model.ts

```diff
@@ -170,7 +174,13 @@ export abstract class Model {
     if (options && options.ignoreUnknownProperties === false) {
       obj = {};
       for (const p in this) {
-        obj[p] = this[p];
+        let val = (this as AnyObject)[p];
+        if (val != null && typeof val.toObject === 'function') {
+          val = val.toObject();
+        } else if (val != null && typeof val.toJSON === 'function') {
```

Comment:
I guess `toJSON` is our fallback from the `toObject` method? I.e. why do we want to have this `else if` block?

Reply:
Use `toJSON()` as the fallback